### DISCUSSION TITLE
If TimeToGo value were to be flexible, what would you like to see in the TimeToGo field?

### DISCUSSION THREAD

**Discussion by CaptKrisp** [Louisvdw/dbus-serialbattery#696] [Ideas]
To better clarify this question, here are a few values I think might be of interest to see populated in the TimeToGo field.
1) While charging:
- a) if SoC < SOC_LOW_ALARM value then "time to SOC_LOW_ALARM"
- b) if SoC < SOC_LOW_WARNING value then "time to SOC_LOW_WARNING"
- c) if SoC < Minimum SoC then "time to Minimum SoC"
- d) time to 100%.
2) While discharging:
 - a) if SoC > Minimum SoC then "time to Minimum SoC"
 - b) if SoC > SOC_LOW_WARNING value then "time to SOC_LOW_WARNING"
 - c) if SoC > SOC_LOW_ALARM value then "time to SOC_LOW_ALARM"
3) While charging or discharging:
 - a) time to (and optionally from) values set TIME_TO_SOC_POINTS

Do you like any of these ideas?  

Have any ideas you'd like to share?

**Comment by mr-manuel** [comment]
Just enable Time To SoC in the config file and you have the data you need. No need for changes.

**Comment by CaptKrisp** [comment]
Thanks for the thought @mr-manuel ... however, that doesn't address the question posed.
As the contributor to the original code for Time To SoC I'm fully aware of it's function.  That code does nothing to display those said times/values in the "TimeToGo" field on the console or VRM so are useful only if utilizing an external system.... such as node-red.

This question is posed to determine interest in placing flexible "target" values in the TimeToGo field.

**Reply by mr-manuel** [reply]
As per official internal Victron Energy documentation (I unfortunately cannot link to) the Time to Go is calculated only on discharging. We already are calculating it also when we are charging and I was thinking about to remove this again, even if I added this. With the new GUIv2 from Victron this can probably cause problems.

For this there are the Time to Soc values you can flexible choose. Therefore I don't like any of your ideas also because you want to integrate something that is already existing in another field. Also if you want to use Node-RED you can calculate the time you need by yourself, since you have all data available you need to do so.

**Reply by CaptKrisp** [reply]
To mention that the official documentation specifies value calculated only on discharge then immediately state that the value is also calculated on charge nullifies your first statement.

I hadn't heard of GUIv2.  I would be interested to learn more about that.

It was obvious from your first response that you do not favor any of the ideas I mentioned but that doesn't mean other people would not be interested.

Again, I am the creator of the code for the various TimeToSoc points so I am perfectly aware of their availability/function/purpose.
And again.... none of those values are available in the GUI nor VRM.

Not everyone who might be interested in seeing the various time-to-soc in GUI and VRM is capable of creating Node-red flows... just as not everyone using VenusOS is capable of creating the entire OS.  Additionally, even if those various time-to-soc values are displayed on a Node-red dashboard it isn't always easily available outside of a LAN... particularly if using satellite internet service.

Furthermore, your responses have indicated that you don't actually understand the intent or proposal of this thread.  Perhaps it would be better if you asked questions to help you fully understand the intent instead of an immediate attempt to shutdown the concept.

BTW, several months ago I had already implemented this exact idea in my local copy of the code.  It's not difficult to do... at least not for someone who has 40+ years programming experience (such as myself).

Lastly, the only downside to this type of functionality in the current TimeToGo field is that there is no way to know what is the current TimeToGo target.  This will easily cause confusion unless there is a way to display what the TTG target is.

**Reply by mr-manuel** [reply]
We added some of the most used TTS points in the GUI, but you are right. You don't see them directly in the VRM, but in the remote cobsole in VRM. See https://github.com/Louisvdw/dbus-serialbattery/blob/master/etc/dbus-serialbattery/qml/PageBattery.qml#L193-L227

**Reply by CaptKrisp** [reply]
Interesting.  I see those lines in my installed version but am not able to locate those values on any GUI page.

**Reply by mr-manuel** [reply]
Have you enabled the calculation for this SoC points in the config? Do you see the values for this SoC points in the dbus-spy?

![grafik](https://github.com/Louisvdw/dbus-serialbattery/assets/12592765/5966f26c-d974-42a3-9484-04699a176f06)

**Reply by CaptKrisp** [reply]
Yes, they are enabled in the config.ini file and I do see them in dbus-spy for the physical devices but not for Battery Aggregator.
TIME_TO_SOC_POINTS = 100, 90, 80, 50, 30, 20, 10, 0

**Reply by CaptKrisp** [reply]
It might be important to note that my 12V system is running Venus OS 3.00-42 and 48V system running 3.10-2.
If I update either system to 3.10~8 then the log (/var/log/dbus-serialbattery.ttyUSB0/current) repeatedly gives message of:

`@40000000648bd57014556414 Traceback (most recent call last):
@40000000648bd57014557f6c   File "/opt/victronenergy/dbus-serialbattery/dbus-serialbattery.py", line 25, in <module>
@40000000648bd570145592f4     from bms.daly import Daly
@40000000648bd57014559ac4 ModuleNotFoundError: No module named 'bms'
`
